src/vs/workbench/contrib/search/browser/searchModel.ts

```diff
@@ -1946,6 +1975,103 @@ export class SearchModel extends Disposable {
 		return this._searchResult;
 	}
 
+	private async runFileQueries(includes: (string)[], token: CancellationToken, textQuery: ITextQuery): Promise<URI[]> {
+		const promises = includes.map(include => {
+			const query: IFileQuery = {
+				type: QueryType.File,
+				filePattern: include,
+				folderQueries: textQuery.folderQueries
+			};
+			return this.searchService.fileSearch(
+				query,
+				token
+			);
+		});
+		const result = (await Promise.all(promises)).map(sc => sc.results.map(fm => fm.resource)).flat();
+		const uris = new ResourceSet(result, uri => this.uriIdentityService.extUri.getComparisonKey(uri));
+		return Array.from(uris.keys());
+	}
+	private async getClosedNotebookResults(textQuery: ITextQuery, scannedFiles: ResourceSet, token: CancellationToken): Promise<{ results: ResourceMap<IFileMatchWithCells | null>; limitHit: boolean }> {
+		const infoProviders = this.notebookService.getContributedNotebookTypes();
+		const includes = infoProviders.flatMap(
+			(provider) => {
+				return provider.selectors.map((selector) => {
+					const globPattern = (selector as INotebookExclusiveDocumentFilter).include || selector as IRelativePattern | string;
+					return globPattern.toString();
+				}
+				);
+			}
+		);
+
+		const results = new ResourceMap<IFileMatchWithCells | null>(uri => this.uriIdentityService.extUri.getComparisonKey(uri));
+
+		const start = Date.now();
+
+		const filesToScan = await this.runFileQueries(includes, token, textQuery);
```

Comment:
This would be better if we could stream file results back, alas file search does not support it. But you could structure it so that you can handle results from each query as they complete, rather than wait for all queries to finish